src/tribler-common/tribler_common/sentry_reporter/sentry_tools.py

```diff
@@ -47,30 +47,30 @@ def parse_stacktrace(stacktrace):
     return [line for line in stacktrace.split('\n') if line]
 
 
-def first(l, default=''):
+def first(iterable, default=''):
     """ Safe return the first item in a list.
 
     Args:
-        l: a list
+        iterable: an iterable sequence
         default: a default value to return in case no items in list.
 
     Returns:
         The first item in list.
     """
-    return l[0] if l else default
+    return iterable[0] if iterable else default
```

Comment:
Actually, I expected the description to be removed too, to increase readability.
Contrast:

```python

def first(iterable, default=''):
    """ Safe return the first item in a list.

    Args:

        l: a list
        iterable: an iterable sequence
        default: a default value to return in case no items in list.

    Returns:

        The first item in list.

    """
    return iterable[0] if iterable else default
```

to

```python

def first_item(iterable, default=''):
    return iterable[0] if iterable else default
```

of even

```python

def first_item(iterable, default=''):
    # Return the first item in the iterable or substitute it with a default value
    return iterable[0] if iterable else default
```



The second one is much easier to parse visually. [The code *is* the documentation in this case](https://en.wikipedia.org/wiki/Self-documenting_code).

When reading the 9-lines-comment version, it took me a disproportionate amount of mental energy to skip through all the comments and find the single line of code under all those comments. The thing is just a non-method, simple function (:+1:  for functional style!) that "returns the first item in the list or substitutes it with a default value". That's a human-readable description that has fewer words then there are lines of code in the function!). Python was *designed* to be readable w/o lengthy comments in such trivial cases.

@drew2a , your code is *beautiful* and *disciplined* but, you know, **sometimes less is more**.

Reply:
`Self-documenting code` is a beautiful concept. 

I often hear this sentence here in Tribler.
Like "we have no documentation, but our code is the documentation". 
It doesn't work.

That's why sometimes I write *more* comments than needed. 
Because to skip comment is simpler than imagine absence documentation.

But agree, we should keep all in balance.

Reply:
It is probably a bikeshedding, but `iterable` is not a good name here because it means something specific in Python: it is an object with `__iter__()` method for which you can do `iter(obj)` to get an iterator. Something like `items` looks less ambiguous.

Reply:
@kozlovsky @ichorid renamed.